### ISSUE TITLE
unbundle operation overrides the path of the lib directory with a bad value

### ISSUE THREAD

**Issue by lmaze** [jspm/jspm-cli#1706]
Hi there,

I think I found what looks like a bug in the latest beta version (0.17 beta12).

I have the following configuration in my ``package.json`` file :
```
"jspm": {
        "name": "my-project",
        "directories": {
            "lib": "",
            "baseURL": "src/main",
            "packages": "src/main/common/components/vendor"
        },
		.....
}
```

I have now no ``lib`` directory because I have several directories containing my sources at the root of my project (``src/main``).

When I run the bundle command ``jspm bundle my-project.js src/main/dist/app-bundle.js --inject`` everything works well : the bundle is generated correctly and I can use it as expected.

But when I run the unbundle command ``jspm unbundle``, the path of the ``lib`` directory in the ``package.json`` file is overriden with the following value :
```
"jspm": {
        "name": "my-project",
        "directories": {
            "lib": "src/main/src",
            "baseURL": "src/main",
            "packages": "src/main/common/components/vendor"
        },
		.....
}
```

This value ``src/main/src`` for the ``lib`` directory is incorrect and the next bundle operation fail logically.

Why the value of the ``lib`` directory is changed during the unbundle process?

**Comment by guybedford** [comment]
This could just as well happen when running `jspm init`, it just happens to be happening on unbundle.

The reason is that an empty lib property is not actually supported, so it is reverting to the default value. It may be possible to consider workflows for empty or `.` lib values in future.

**Comment by lmaze** [comment]
OK, thanks for your quick reply !

As I understand, all js files should be in a directory that should not be the root. It seems that I fell in the wrong case ;-) Reverting to the default value is perhaps a bit brutal (the default value will lead to a bad situation anyway).

Is there a particular reason to not support this case? Or is it just an unexpected use case?

**Comment by guybedford** [comment]
The ability to support this will be in the next beta release now.

**Comment by lmaze** [comment]
Thank you very much, it's so great !

I have tested the support by upgrading to the latest commit available in the 0.17 branch and there seems to be some problems.

**1st problem:**
It's not important but when I run a command "jspm" there is always this warning:
```
warn directories.lib in the package.json must be dans le baseURL for paths to resolve Correctly.
```
If the feature is officially supported, this warning may be unnecessary.

**2nd problem:**
In the "jspm.config.js" file, a new section "nodeConfig" is added but the path generated is incorrect. In my case, the generated code was:
```
nodeConfig: {
    "paths": {
      "my-app/": "..\\../"
    }
}
```
So when I launch a jspm command, I have the following warning:
```
Warn Paths configuration" my-app / "->" .. \ ../ "is not valid in Node, falling back to directories.lib default value.
```

**3rd issue:**
More importantly, when I run the "jspm bundle" command, I get the following error:
```
Err Error: C Plugin: /my-site/my-app/src/main/plugin-babel.js can not load Itself, sure make it is excluded from Any wildcard meta configuration via a custom loader: false rule.
  Error loading C: /my-site/my-app/src/main/plugin-babel.js
```

It seems that the new section of the "jspm.config.js" file :
```
devConfig: {
    "map": {
      "plugin-babel": "npm:systemjs-plugin-babel@0.0.10"
    }
  }
```
is not properly taken into account.

Maybe I made a configuration error, but it worked well in beta version 14. My configuration is also very simple.

**Comment by guybedford** [comment]
Thanks, these should all be fixed on the main branch now... will be posting out a release soon.

**Comment by guybedford** [comment]
Released in jspm 0.17.0-beta.15.
